.github/workflows/test.yml

```diff
@@ -10,16 +10,35 @@ jobs:
   test:
     environment: test
     runs-on: ubuntu-22.04
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     steps:
       - name: git checkout
         uses: actions/checkout@v3
       - name: Check Nixpkgs input
         uses: DeterminateSystems/flake-checker-action@main
         with:
           fail-mode: true
+      - name: Restore Lychee cache
+        uses: actions/cache@v3
+        with:
+          path: .lycheecache
+          key: cache-lychee-${{ github.sha }}
```

Comment:
If it's just cached on the sha, the cache will get busted every time you push something. The only time it will hit the cache is if you need to restart the GHA job. Doesn't sound very useful...